File: rust/tests/executions_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_executions_execution_runtime_lineage_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .executions
        .execution_runtime_lineage(
            &"project_id".to_string(),
            &"trace_id".to_string(),
            &ExecutionRuntimeLineageQueryRequest {
                account_id: "account_id".to_string(),
                inference_after: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "GET",
        "/v1/projects/project_id/executions/trace_id/runtime",
        Some(HashMap::from([(
            "account_id".to_string(),
            json!("account_id"),
        )])),
        1,
    )
    .await
    .unwrap();
}

File: ruby/test/wire/executions_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ExecutionsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_executions_execution_runtime_lineage_with_wiremock
    test_id = "executions.execution_runtime_lineage.0"

    @client.executions.execution_runtime_lineage(
      project_id: "project_id",
      trace_id: "trace_id",
      account_id: "account_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "executions.execution_runtime_lineage.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/executions/trace_id/runtime",
      query_params: { "account_id" => "account_id" },
      expected: 1
    )
  end
end
